usb/usb_host_msc/src/msc_host.c

```diff
@@ -514,11 +537,20 @@ esp_err_t msc_host_write_sector(msc_host_device_handle_t device, size_t sector,
     return scsi_cmd_write10(dev, data, sector, 1, dev->disk.block_size);
 }
 
-esp_err_t msc_host_handle_events(uint32_t timeout_ms)
+static void copy_string_desc(wchar_t *dest, const usb_str_desc_t *src)
 {
-    MSC_RETURN_ON_FALSE(s_msc_driver != NULL, ESP_ERR_INVALID_STATE);
-
-    return usb_host_client_handle_events(s_msc_driver->client_handle, pdMS_TO_TICKS(timeout_ms));
+    if (dest == NULL) {
+        return;
+    }
+    if (src != NULL) {
+        size_t len = MIN((src->bLength - USB_STANDARD_DESC_SIZE) / 2, MSC_STR_DESC_SIZE - 1);
+        wcsncpy(dest, src->wData, len);
```

Comment:
How do you deal with unaligned access warning here? 
Or you do not have any?

Reply:
I suppressed this warning. It is not relevant as the descriptor is always in memory that supports unaligned access